.github/workflows/build.yml

```diff
@@ -308,3 +310,195 @@ jobs:
+
+          // Get all tags and keep the oldest one starting by v
+          let oldest_tag = { name: 'v0.0.0' };
+
+          try {
+            const tags = await github.rest.repos.listTags({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            // Keep latest tag
+            for (const tag of tags.data) {
+              if (tag.name.startsWith('v')) {
+                if (tag.name > oldest_tag.name) {
+                  oldest_tag = tag;
+                }
+              }
+            }
+          } catch (err) {
+            console.log(err);
+          }
+          console.log(`Previous release has tag ${oldest_tag.name} → ${oldest_tag.commit.sha}`);
+
+          // Get current date with format %Y-%m-%d
+          const today = new Date();
+          var ISOToday = today.toISOString().split('T')[0];
+          console.log(`Today: ${ISOToday}`);
+
+          // Count all commits between HEAD and oldest tag
+          let distance = 0;
+          try {
+            let page = 1;
+            let found = false;
+            do {
+              const response = await github.rest.repos.listCommits({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                per_page: 100,
+                page: page,
+              });
+              response.data.forEach(commit => {
+                if (commit.sha === oldest_tag.commit.sha) {
+                  found = true;
+                } else {
+                  if (found !== true) distance++;
+                }
+              });
+              page++;
+            } while (found === false && response.length > 0);
+            console.log(`Found ${distance} commits`);
+          } catch (err) {
+            console.log(err);
+          }
+
+          // Zip a zip file from dist directory
+          console.log(`nightly_${distance}_${ref_head.data.object.sha.substring(0,8)}.zip`);
+
+          let release_name = `nightly_${distance}_${ref_head.data.object.sha.substring(0,8)}` + '.zip';
+
+          fs.renameSync('dist.zip', release_name, function (err) {
+            if (err) throw err;
+            console.log(`dist.zip was renamed to {release_name}`);
+          });
+
+          // Upload the zip file to GitHub
+          let uploadedAsset = null;
+          try {
+            uploadedAsset = await github.rest.repos.uploadReleaseAsset({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              release_id: release.data.id,
+              name: release_name,
+              data: fs.readFileSync(release_name),
+              headers: {
+                'content-type': 'application/zip',
+              },
+            });
+          } catch (err) {
+            console.log(err);
+          }
+          console.log(`Uploaded ${uploadedAsset.data.name} to GitHub`);
```

Comment:
move this log into the try/catch clause, we do not want to display it if an error occurred

Reply:
done

Reply:
in fact this log is useless, this string is already in job output